src/devices/bluetooth.c

```diff
@@ -206,16 +232,31 @@ int bt_init_connection(DeviceConfig *config)
          * factory level (9600).  This ensures things go slow enough for the
          * HC-06 processor + code to handle it.
          */
+        int baud = bt_find_working_baud(config, btConfig);
+        switch (baud) {
+        case 0:
+                pr_info("BT: Failed to communicate with device.  Assuming it "
+                        "is not connected.\r\n");
+                g_bluetooth_status = BT_STATUS_ERROR;
+                return DEVICE_INIT_FAIL;
+        case 9600:
```

Comment:
I think you also want to check if the current baud is BT_AT_CMD_BAUD then you want to fall through and assume it's already configured?